components/brave_ads/core/internal/serving/eligible_ads/pipelines/new_tab_page_ads/eligible_new_tab_page_ads_v2.cc

```diff
@@ -24,6 +26,19 @@
 
 namespace brave_ads {
 
+namespace {
+
+void ApplyConditionMatcher(CreativeNewTabPageAdList& creative_ads) {
```

Comment:
Remind me again, this is used for rewards only correct?

Reply:
We call MatchConditions for non-Rewards users and both MatchConditions and ApplyConditionMatcher for Rewards users.